pallets/tinlake-investor-pool/src/lib.rs

```diff
@@ -751,6 +882,127 @@ pub mod pallet {
 			};
 		}
 
+		pub(crate) fn calculate_collect(
+			loc: TrancheLocator<T::PoolId, T::TrancheId>,
+			order: UserOrder<T::Balance, T::EpochId>,
+			pool: PoolDetails<T::AccountId, T::CurrencyId, T::EpochId, T::Balance, T::InterestRate>,
+			end_epoch: T::EpochId,
+		) -> Result<OutstandingCollections<T::Balance>, DispatchError> {
+			// No collect possible in this epoch
+			if order.epoch == pool.current_epoch {
+				return Ok(OutstandingCollections {
+					payout_currency_amount: Zero::zero(),
+					payout_token_amount: Zero::zero(),
+					remaining_supply_currency: order.supply,
+					remaining_redeem_token: order.redeem,
+				});
+			}
+
+			// It is only possible to collect epochs which are already over
+			let parse_until_epoch = end_epoch.min(pool.last_epoch_executed);
```

Comment:
This value is only used on line 914 so I would move it just before that block.

Also, the name is misleading since it's not parsing anything, it just represents the upper epoch bound we are interested in:

 ```suggestion
			// It is only possible to collect epochs which are already over
			let end_epoch = end_epoch.min(pool.last_epoch_executed);
```